src/main/java/com/coherent/aqa/java/training/web/korobeynik/driver/Driver.java

```diff
@@ -2,27 +2,56 @@
 
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
 
 public final class Driver {
-    private static volatile WebDriver driver;
+    private static WebDriver webDriver;
+    private static volatile Driver driver;
 
     private Driver() {
-        driver.manage().window().maximize();
     }
 
-    public synchronized static WebDriver getDriver() {
-            if (driver == null) {
-                driver = new ChromeDriver();
-            }
-            return driver;
+    public synchronized static Driver getDriver() {
+        if (driver == null) {
+            driver = new Driver();
         }
+        return driver;
+    }
+
+    public WebDriver openRemoteDriver(DriverOptions driverOption) {
+        DesiredCapabilities dc = new DesiredCapabilities();
+        switch (driverOption) {
```

Comment:
instead of switch-case let's get browserName from properties
grid url should be also moved to properties